_posts/2017/2017-12-31-truthiness-in-conditionals.md

```diff
@@ -0,0 +1,59 @@
+---
+title: 'Truthiness in Conditionals'
+description: 'if not not nil?'
+layout: article
+category: ruby
+image:
+  base: '2017/truthiness-in-conditionals'
+  alt: 'Truth'
+  source: 'https://unsplash.com/photos/Aa2h3S3E47k'
+  credit: 'Jarrod Fitzgearlds'
+---
+
+Ruby’s conditional syntax is ‘truthy’, meaning that any statement in a conditional that evaluates to `nil` is considered to be equivalent to `false` and anything not-`nil` can be considered to be `true`.
+
+
+## Instead of…
+
+...overcomplicating your conditions.
+
+```ruby
+# Example 1
+unless something.nil?
+  # do something
+end
+
+# Example 2
+if !something.nil?
+  # do something
+end
+
+# Example 3
+if !!something
+  # do something
+end
+```
+
+
+## Use…
+
+```ruby
+# Instead of Examples 1,2 & 3
+if something
+  # do something
+end
+```
+
+
+## But why?
+
+Performing a `#nil?` check as part of a statement in a negative conditional, as in the first two examples (`unless` or `if !`), is often redundant. Any non-`nil` value is considered to be ‘truthy’.
```

Comment:
*is often redundant as any  non-`nil` value is `truthy`. Therefore, you might as well perform the check as part of a statement in a positive conditional.

I still think there's something that's missing that is the: "oh, of course. That's why it's redundant".

Reply:
Any `nil` value is 'false-y', so you can achieve the same result with a positive conditional and no need for the `#nil?` check.

Reply:
Any nil value is 'falsey', so you can achieve the same result with a positive conditional and no #nil? check.